examples/plotting/plot_rectangle.py

```diff
@@ -0,0 +1,100 @@
+"""
+==========================================================
+Drawing a Rectangle on a map
+==========================================================
+
+In this example we are going to look at how we can plot rectangles on maps
+while describing the dimensions using different methods.
+"""
+
+################################################################################
+# Fundamentally, we will be using the `~sunpy.map.GenericMap.draw_rectangle` method to
+# draw the rectangles. Its function signature is like so:
+#
+# `draw_rectangle(bottom_left, *, width: Unit(‘deg’) = None,
+# height: Unit(‘deg’) = None, axes=None, top_right=None, **kwargs)`
+#
+# All of the extra kwargs are passed on to the `~matplotlib.patches.Rectangle` instance,
+# which allows us to modify the color, linestyle, etc. of the plotted rectangle.
+# A full overview of available options can be found in the Rectangle documentation.
+#
+# Therefore, the different forms in which an input can be provided to the function are:
+#
+# * `~astropy.coordinates.SkyCoord` ((x1, x2), (y1, y2))
+# * `~astropy.coordinates.SkyCoord` bottom_left, `~astropy.coordinates.SkyCoord` top_right
+# * `~astropy.coordinates.SkyCoord` bottom_left, `~astropy.units.Quantity` width, `~astropy.units.Quantity` height
+#
+# Note that from SunPy v2.1, every argument to `~sunpy.map.GenericMap.draw_rectangle` apart from
+# 'bottom_left' (i.e. 'top_right', 'width' and 'height') should be passed as a keyword argument.
+#
+# We also have a fourth option - to specify the dimensions in terms of pixel coordinates.
```

Comment:
I don't think this needs to be explicitly stated here. Or at the very least, this should be broken up and interspersed with the code below.